docs/error_messages.md

```diff
@@ -0,0 +1,368 @@
+
+## Resolution_IrreplaceableLocalRef
+
+**Description:** Unable to replace a local reference
+
+### Suggested User Actions
+
+1. This is likely a cycle of local references, which the import tool is not able to resolve. Check your MCF files for potential cycles.
+
+## Resolution_UnassignableNodeDcid
+
+**Description:** Unable to assign DCID due to unresolved local reference
+
+### Suggested User Actions
+
+1. See actions for `Resolution_IrreplaceableLocalRef`
+
+## Resolution_DcidAssignmentFailure_
+
+**Description:** The node could not be assigned a DCID based on the data available. DCID can be generated for StatVarObs, a legacy population type (type ends with `Population`), a legacy observation type (type ends with `Observation` and is not `StatVarObservation`), or if there is an external ID resolver provided.
+
+**Suffix Description:** The node could not be assigned a DCID based on the data available. DCID can be generated for StatVarObs, a legacy population type (type ends with `Population`), a legacy observation type (type ends with `Observation` and is not `StatVarObservation`), or if there is an external ID resolver provided.
+
+### Suggested User Actions
+
+1. If none of the conditions in the description apply to your node, provide a non-empty DCID for the node.
+
+## Resolution_OrphanLocalReference_
+
+**Description:** The local ID of the node is missing from the entire sub-graph resolved by the import tool.
+
+**Suffix Description:** The local ID of the node is missing from the entire sub-graph resolved by the import tool.
+
+## Resolution_ReferenceToFailedNode_
+
+**Description:** The reference was resolved, but to a failed node, therefore, this node also failed to resolve
+
+**Suffix Description:** The reference was resolved, but to a failed node, therefore, this node also failed to resolve
+
+### Suggested User Actions
+
+1. Check the logs for the failure of the node identified the error message and address that issue.
+
+## Sanity_InconsistentSvObsValues
+
+**Description:** Found different values provided for the same StatVarObservation
+
+### Suggested User Actions
+
+1. Check for any duplication in the input CSV.
+
+## Sanity_SameDcidForDifferentStatVars
+
+**Description:** The same curated DCID was found for different StatVars.
+
+### Suggested User Actions
+
+1. Ensure that StatVars have distinct names.
+
+## Sanity_DifferentDcidsForSameStatVar
+
+**Description:** Found different curated IDs for same StatVar
+
+### Suggested User Actions
+
+1. Ensure that the StatVars have consistent DCIDs.
+
+## CSV_HeaderFailure
+
+**Description:** Unable to parse header from CSV file
+
+### Suggested User Actions
+
+1. Check for malformation in the headers of input CSV files.
+
+## CSV_TmcfCheckFailure
+
+**Description:** There was a fatal sanity error in TMCF
+
+### Suggested User Actions
+
+1. Check counter messages that start with Sanity_ for FATAL error level
+
+## Sanity_TmcfMissingEntityDef
```

Comment:
The counter names have a category to them.  

TMCF/MCF/CSV:  counters from parsing of corresponding files
Sanity: counters from basic sanity check of DC nodes.
Existence: counters from existence check against Data Commons
Resolution: counters from resolution.

We could probably put them under different categories.

Generally, I feel, let's hand edit the `error_messages.md` first, and then turn it into code/config.